pkg/provisioningv2/rke2/planner/registry.go

```diff
@@ -34,21 +34,20 @@ func toRegistryConfig(secrets v1.SecretCache, runtime, namespace string, registr
 
 	for registryName, config := range registry.Configs {
 		registryConfig := &registryConfig{}
-		if config.TLSSecretName == "" && config.InsecureSkipVerify {
+		if config.InsecureSkipVerify || config.TLSSecretName != "" || len(config.CABundle) > 0 {
```

Comment:
does this need to be in an if check at all if its getting set for all conditions?

Reply:
We only want to set the `registryConfig.TLS` if it has any non-empty config because this gets passed to RKE2/K3S during provisioning. If there is an empty TLS config when RKE2/K3S is started, then command line args validation there will fail.